Employees/Services/Log.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using Employees.Services;
+using Serilog;
+
+namespace Employees.Service
+{
+    public class Log : ILogs
+    {
+        public string FullCrashLogPath { get; set; }
+        public string FullEmployeesCreatedLogPath { get; set; }
+        public Log(string crashLogFile, string employeesCreatedLogFile)
+        {
+            FullCrashLogPath = crashLogFile;
+            FullEmployeesCreatedLogPath = employeesCreatedLogFile;
+        }
+        public void CrashLogs()
+        {
+            string finalCrashLogPath = Path.GetFullPath(Path.Combine(FullCrashLogPath, "CrashLogs.txt"));
+            using (var log = new LoggerConfiguration()
+                        .WriteTo.File(finalCrashLogPath, rollingInterval: RollingInterval.Day)
+                        .CreateLogger())
+            {
+                AppDomain.CurrentDomain.FirstChanceException += (sender, EventArgs) =>
+                log.Error(" This error: {EventArgs}, {Now}", EventArgs.Exception.ToString(), DateTime.Now); 
+            }
+        }
+
+        public void EmployeesCreatedLogs(string newEmployeeLogged)
+        {
+            string finalEmployeeLogPath = Path.GetFullPath(Path.Combine(FullEmployeesCreatedLogPath, "EmployeeLogsText.txt"));
+            using (var log = new LoggerConfiguration()
+                        .WriteTo.File(finalEmployeeLogPath, rollingInterval: RollingInterval.Day)
+                        .CreateLogger())
+            {
+                log.Information("Employee added {newEmployeeLogged} when {Now}", newEmployeeLogged, DateTime.Now);
+            }
+        }
+    }
+}
```

Comment:
@AalaOsman1 This is an example of the interface segregation principle (the I in SOLID). You have this interface at the moment:

```csharp
public interface ILogs
{
    void EmployeesCreatedLogs(string NewEmployeeLogged);
    void CrashLogs();
}
```

When you implement that to, for example, write logs to files in folders, you need to wire up both kinds of log to go to a file. Now imagine your app is in production, and the "employee created" log is something you might go to occasionally if you need to check something but the "crash log" needs to be dealt with urgently. You decide to stick with writing a log file for the "employee created" log but implement a service like [Exceptionless](https://exceptionless.com/) for your crash logs. 

To implement the logging to Exceptionless you _have to_ implement the `EmployeesCreatedLogs` method even though you have no intention of sending those logs to Exceptionless. Now imagine your app has 20 kinds of log and the task is much bigger, and you're writing a lot of unnecessary code - that's what the interface segregation principle is trying to fix.

A different way to approach it would be something like:

```csharp
public interface ILog 
{
    public bool CanLog(LogType logType);
    public void Log(string message);
}

public enum LogType 
{
    EmployeeCreated,
    Crash
}
```

You'd implement it twice:

```csharp
public class EmployeeCreatedLog : ILog {
    public bool CanLog(LogType logtype) => logType == LogType.EmployeeCreated;
    public void Log(string message) {
       // code to write to file
    }
}

public class CrashLog : ILog {
    public bool CanLog(LogType logtype) => logType == LogType.Crash;
    public void Log(string message) {
       // code to write to Exceptionless
    }
}
```

Then your application can have a collection of all the `ILog` implementations and, when it needs to log, it can do something like:

```csharp
foreach (ILog log in _logs)
{
   if (log.CanLog(LogType.EmployeeCreated)) 
   {
       log.Log("Employee xxx was created");
   }
}
```

Typically what we do is use logging tools like [Log4Net](https://logging.apache.org/log4net/) or [SeriLog](https://serilog.net/) that implement that kind of thing for us, but that explains the reasoning behind it.